tests/test_fips.py

```diff
@@ -139,22 +139,50 @@ def openssl_fips_hashes_test_fnct(container_per_test: ContainerData) -> None:
     all non-fips hash algorithms fail.
 
     """
+    c = container_per_test.connection
     for digest in NONFIPS_DIGESTS:
-        cmd = container_per_test.connection.run(f"openssl {digest} /dev/null")
+        cmd = c.run(f"openssl {digest} /dev/null")
         assert cmd.rc != 0
         assert (
             "is not a known digest" in cmd.stderr
             or "Error setting digest" in cmd.stderr
         )
 
     for digest in FIPS_DIGESTS:
-        dev_null_digest = container_per_test.connection.check_output(
-            f"openssl {digest} /dev/null"
-        )
+        dev_null_digest = c.check_output(f"openssl {digest} /dev/null")
         assert f"= {NULL_DIGESTS[digest]}" in dev_null_digest, (
             f"unexpected digest of hash {digest}: {dev_null_digest}"
         )
 
+    if OS_VERSION == "15.6":
+        # Removal of default version openSSL3
+        c.check_output("zypper --non-interactive rm --clean-deps openssl")
+
+        # Download and extract the openSSL1.1 legacy version
+        arch = LOCALHOST.system_info.arch
+        legacy_repourl = f"https://download.suse.de/ibs/SUSE/Products/SLE-Module-Legacy/15-SP6/{arch}/product"
```

Comment:
BIG WARNING SIGN.

The test subject changes! Here we would be testing the openssl binary from a place outside of the container and the BCI repository. Is this really what we want to do? :thinking:

Reply:
The ticket states `Add similar workaround to BCI that is being done at https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/security/openssl_misc_utils.pm#L64 for SLED and SLE RT.`, so I assumed that was a right place where to get the binary from...

Maybe Timo Jyrinki should shed some light here?